plugins/plugin-chart-table/src/transformProps.ts

```diff
@@ -24,14 +24,52 @@ import {
   getTimeFormatter,
   smartDateFormatter,
   getTimeFormatterForGranularity,
+  TimeFormatFunction,
+  TimeFormatter,
 } from '@superset-ui/time-format';
 
 import isEqualArray from './utils/isEqualArray';
 import { TableChartProps, TableChartTransformedProps, DataType, DataColumnMeta } from './types';
 
 const { PERCENT_3_POINT } = NumberFormats;
 const TIME_COLUMN = '__timestamp';
-const toString = (x: DataRecordValue) => String(x);
+const REGEXP_TIMESTAMP_NO_TIMEZONE = /T(\d{2}:){2}\d{2}$/;
+
+/**
+ * Extended Date object with custom formatter.
+ * By default `String(date)` returns the raw input.
+ */
+export class DateWithFormatter extends Date {
+  formatter: TimeFormatFunction;
+
+  input: DataRecordValue;
+
+  constructor(
+    input: DataRecordValue,
+    { formatter = String, forceUTC = true }: { formatter?: TimeFormatFunction; forceUTC?: boolean },
+  ) {
+    let value = input;
+
+    // force UTC time for timestamps without a timezone
+    if (forceUTC && typeof value === 'string' && REGEXP_TIMESTAMP_NO_TIMEZONE.test(value)) {
+      value = `${value}Z`;
+    }
+
+    // react-table sorter expects all records to have Date type
+    // `new Date()` is capable of handling null/boolean/undefined, etc;
+    super(value as string);
+
+    this.input = input;
+    this.formatter = formatter;
+  }
+
+  toString(): string {
+    if (this.formatter === String) {
+      return String(this.input);
```

Comment:
maybe this should be `return this.formatter(this.input);`?

Reply:
It's basically the same thing?